packages/core/src/dialects/db2/query-generator-typescript.ts

```diff
@@ -228,4 +229,12 @@ export class Db2QueryGeneratorTypeScript extends AbstractQueryGenerator {
 
     return fragment;
   }
+
+  createSavepointQuery(savepointName: string): string {
+    return `SAVEPOINT ${this.quoteIdentifier(savepointName)} ON ROLLBACK RETAIN CURSORS`;
```

Comment:
the retain cursors part is new. Should it be enabled by default? Could be an option instead

Reply:
Yes, this is a required as part of the [SAVEPOINT statement](https://www.ibm.com/docs/en/db2/11.5?topic=statements-savepoint).

I could add the `ON ROLLBACK RETAIN LOCKS` option but I couldn't see how this could be utilized.